crates/astria-conductor/src/conductor.rs

```diff
@@ -10,122 +18,239 @@ use tokio::{
-                //       handlers should have the same bias.
+                // FIXME: The bias should only be on the signal channels. The two handlers should have the same bias.
                 biased;
 
                 _ = stop_rx.changed() => {
                     info!("shutting down conductor");
-                    if let Some(e) = driver_tx.send(DriverCommand::Shutdown).err() {
-                        error!(
-                            error.msg = %e,
-                            error.cause = ?e,
-                            "error sending Shutdown command to driver"
-                        );
-                    }
                     break;
                 }
 
                 _ = reload_rx.changed() => {
                     info!("reloading is currently not implemented");
                 }
-                // request new blocks every X seconds
-                _ = interval.tick() => {
-                    if let Some(e) = driver_tx.send(DriverCommand::GetNewBlocks).err() {
-                        // the only error that can happen here is SendError which occurs
-                        // if the driver's receiver channel is dropped
-                        error!(
-                            error.msg = %e,
-                            error.cause = ?e,
-                            "error sending GetNewBlocks command to driver"
-                        );
-                        break;
-                    }
-                }
 
-                res = &mut driver => {
+                res = &mut sync_done, if !sync_done.is_terminated() => {
                     match res {
-                        Ok(Ok(())) => error!("driver task exited unexpectedly"),
-                        Ok(Err(e)) => error!(error.msg = %e, error.cause = ?e, "driver exited with error"),
-                        Err(e) => error!(error.msg = %e, error.cause = ?e, "driver task failed"),
+                        Ok(()) => info!("received sync-complete signal from sequencer reader"),
+                        Err(e) => warn!(error.message = %e, error.cause = ?e, "sync-complete channel failed prematurely"),
                     }
-                    break;
-                }
-
-                res = &mut executor => {
-                    match res {
-                        Ok(Ok(())) => error!("executor task exited unexpectedly"),
-                        Ok(Err(e)) => error!(error.msg = %e, error.cause = ?e, "executor exited with error"),
-                        Err(e) => error!(error.msg = %e, error.cause = ?e, "executor task failed"),
+                    if let Some(data_availability_reader) = data_availability_reader.take() {
+                        info!("starting data availability reader");
+                        tasks.spawn(
+                            Self::DATA_AVAILABILITY,
+                            data_availability_reader.run_until_stopped(),
+                        );
```

Comment:
the sync occurs entirely from the sequencer node instead of from DA? guess this makes sense, since the sequencer nodes should hold all block data (currently). i wonder if syncing from a celestia node could also make sense eventually

Reply:
Yeah sync right now occurs solely from sequencer, I agree that there is a use case for DA sync (quite possibly more useful) in the future. The reason to not have this right now is that we haven't gotten everything configured to run against prod celestia DA, and rollups run by 3rd parties can't connect to our custom celestia network right now.

This is done largely to avoid inheriting instability from prelaunch Celestia.